corehq/messaging/smsbackends/vertex/models.py

```diff
@@ -61,13 +67,31 @@ def populate_params(self, msg_obj):
     def send(self, msg_obj, *args, **kwargs):
         params = self.populate_params(msg_obj)
         resp = requests.get(VERTEX_URL, params=params)
-        self.handle_response(msg_obj, resp.text)
+        self.handle_response(msg_obj, resp.status_code, resp.text)
 
-    def handle_response(self, msg_obj, resp):
+    def handle_response(self, msg_obj, resp_status_code, resp_text):
         # in case of success store the complete response msg which should be like
         # 570737298-2017_05_27 AS-IS. This complete id referred to as "Scheduleid"
         # can be then used to enquire for status of this particular SMS
-        if SUCCESS_RESPONSE_REGEX_MATCHER.match(resp):
-            msg_obj.backend_message_id = resp
+        if SUCCESS_RESPONSE_REGEX_MATCHER.match(resp_text):
+            msg_obj.backend_message_id = resp_text
         else:
-            msg_obj.set_system_error(resp)
+            self.handle_failure(msg_obj, resp_status_code, resp_text)
+
+    def handle_failure(self, msg_obj, resp_status_code, error_message):
+        error_code, text = GATEWAY_ERROR_MESSAGE_REGEX_MATCHER.match(error_message).groups()
+        if error_code and error_code == INCORRECT_MOBILE_NUMBER_CODE:
+            msg_obj.set_system_error(SMS.ERROR_INVALID_DESTINATION_NUMBER)
+        elif (
+            error_code in GATEWAY_ERROR_CODES or
+            error_message in GATEWAY_ERROR_MESSAGES
+        ):
+            msg_obj.set_system_error(SMS.ERROR_TOO_MANY_UNSUCCESSFUL_ATTEMPTS)
+            raise VertexBackendException(error_message)
```

Comment:
raising the exception causes it to be retried which we can avoid for these known errors by just notifying exception like we do here: https://github.com/dimagi/commcare-hq/blob/master/corehq/messaging/smsbackends/icds_nic/models.py#L118